docs/mutability.md

```diff
@@ -0,0 +1,63 @@
+# Mutability in Substratus
+
+## Models
+
+## Datasets
+
+## Servers
+
+```yaml
+kind: Server
+
+spec:
+
+  params: {} # Change to field triggers redeploy.
+
+  image:
+    name: "" # Change to field triggers redeploy.
+    upload:
+      md5Checksum: "" # Change to field triggers rebuild.
+    git:
+      repo: ""    # Change to field triggers rebuild.
+
+      tag: ""     # Same as branch:
+      branch: ""  # Change to field triggers rebuild.
+                  # Change in git reported in .status.image.synced = false ...
+                  # Rebuild happens automatically.
+
+      commit: ""  # Change to field triggers rebuild. Git not monitored for changes.
```

Comment:
commit doesn't seem to be a thing in the API spec?

Reply:
Not currently, will add that comment